pyccel/ast/numpyext.py

```diff
@@ -416,13 +423,36 @@ def Shape(arg):
         return PythonTuple(*arg.shape)
 
 #==============================================================================
+
 class NumpyImag(PythonImag):
-    """Represents a call to  numpy.real for code generation.
+    """Represents a call to  numpy.imag for code generation.
 
     > a = 1+2j
     > np.imag(a)
     2.0
     """
+    def __new__(cls, arg):
+        if not isinstance(arg.dtype, NativeComplex):
+            dtype=NativeInteger() if isinstance(arg.dtype, NativeBool) else arg.dtype
+            if isinstance(arg, (Literal, Variable)) and arg.rank == 0:
+                return convert_to_literal(0, dtype, arg.precision)
+            return NumpyZeros(arg.shape, dtype=dtype)
```

Comment:
Why do you only return a literal if the argument is of type: `Literal` or `Variable` ?

Reply:
because the imag part of every non-complex scalar is 0, so i took care about just the `dtype`, i suspect `Variable` objects can surprise me but since the rank is set correctly, i think it is fine to return just 0 literal with the correct `dtype` for know.

Reply:
So:
```
np.imag(1+2)
```
Should return:
```
[0]
```
?

Reply:
Returning a literal for every non-complex scalar is exactly right. But that is not what you are doing